packages/medusa/src/strategies/product-export.ts

```diff
@@ -0,0 +1,781 @@
+      },
+    ],
+    [
+      "Variant Length",
+      {
+        accessor: (variant: ProductVariant): string =>
+          variant?.length?.toString() ?? "",
+        entityName: "variant",
+      },
+    ],
+    [
+      "Variant Width",
+      {
+        accessor: (variant: ProductVariant): string =>
+          variant?.width?.toString() ?? "",
+        entityName: "variant",
+      },
+    ],
+    [
+      "Variant Height",
+      {
+        accessor: (variant: ProductVariant): string =>
+          variant?.height?.toString() ?? "",
+        entityName: "variant",
+      },
+    ],
+    [
+      "Variant HS Code",
+      {
+        accessor: (variant: ProductVariant): string =>
+          variant?.hs_code?.toString() ?? "",
+        entityName: "variant",
+      },
+    ],
+    [
+      "Variant Origin Country",
+      {
+        accessor: (variant: ProductVariant): string =>
+          variant?.origin_country?.toString() ?? "",
+        entityName: "variant",
+      },
+    ],
+    [
+      "Variant Mid Code",
+      {
+        accessor: (variant: ProductVariant): string =>
+          variant?.mid_code?.toString() ?? "",
+        entityName: "variant",
+      },
+    ],
+    [
+      "Variant Material",
+      {
+        accessor: (variant: ProductVariant): string =>
+          variant?.material?.toString() ?? "",
+        entityName: "variant",
+      },
+    ],
+  ])
+
+  private readonly NEWLINE = "\r\n"
+  private readonly DELIMITER_ = ";"
+  private readonly BATCH_SIZE = 50
+
+  constructor({
+    manager,
+    batchJobService,
+    productService,
+    fileService,
+    productRepository,
+  }: InjectedDependencies) {
+    super({
+      manager,
+      batchJobService,
+      productService,
+      fileService,
+      productRepository,
+    })
+
+    this.manager_ = manager
+    this.batchJobService_ = batchJobService
+    this.productService_ = productService
+    this.fileService_ = fileService
+    this.productRepository_ = productRepository
+  }
+
+  async buildTemplate(): Promise<string> {
+    return this.buildHeader()
+  }
+
+  async completeJob(batchJobId: string): Promise<BatchJob> {
+    return await this.atomicPhase_(async (transactionManager) => {
+      const batchJob = await this.batchJobService_
+        .withTransaction(transactionManager)
+        .retrieve(batchJobId)
+
+      if (batchJob.status === BatchJobStatus.COMPLETED) {
+        return batchJob
+      }
+
+      return await this.batchJobService_.complete(batchJob)
+    })
+  }
+
+  async prepareBatchJobForProcessing(
+    batchJobId: string,
+    req: Express.Request
+  ): Promise<BatchJob> {
+    return await this.atomicPhase_(async (transactionManager) => {
+      return await this.batchJobService_
+        .withTransaction(transactionManager)
+        .ready(batchJobId /* batchJob after the above todo */)
+    })
+  }
+
+  async processJob(batchJobId: string): Promise<BatchJob> {
+    let offset = 0
+    let advancementCount = 0
```

Comment:
yes it is when there is a retry, it can start from an already known offset.
`productCount` should not take the offset into consideration, the export is still done base on the input listConfig from the context, and when we retry the context shouldn't have changed and we restart at the last offset